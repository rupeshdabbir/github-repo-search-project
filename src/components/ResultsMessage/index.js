// @flow

import React from 'react';
import { StyleSheet, css } from 'aphrodite/no-important';
import 'suitcss-utils-flex/lib/flex.css';

type Props = {
  pageTotal: number,
  resultsTotal: number,
  searchTerm: string,
};

function ResultsMessage({ searchTerm, resultsTotal, pageTotal }: Props) {
  if (!searchTerm || !pageTotal) {
    return null;
  }

  const [query] = searchTerm.split(' ');

  return (
    <div className="u-flex u-flexJustifyCenter">
      <p className={css(styles.ResultsMessage_text)}>
        Found {resultsTotal} {resultsTotal === 1 ? 'result' : 'results'} for <b>{query}</b>
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  ResultsMessage_text: {
    padding: 6,
    backgroundColor: '#fcf8bb',
    borderRadius: 8,
  },
});

export default ResultsMessage;
