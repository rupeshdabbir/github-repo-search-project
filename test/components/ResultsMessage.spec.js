import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import ResultsMessage from 'components/ResultsMessage';

describe('Component: ResultsMessage', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <ResultsMessage
        searchTerm={'test'}
        pageTotal={10}
        resultsTotal={50}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should handle single results', () => {
    const component = renderer.create(
      <ResultsMessage
        searchTerm={'test'}
        pageTotal={1}
        resultsTotal={1}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
