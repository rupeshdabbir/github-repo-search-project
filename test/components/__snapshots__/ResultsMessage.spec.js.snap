// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResultsMessage should handle single results 1`] = `
<div
  className="u-flex u-flexJustifyCenter"
>
  <p
    className="ResultsMessage_text_lesmrd"
  >
    Found 
    1
     
    result
     for 
    <b>
      test
    </b>
  </p>
</div>
`;

exports[`Component: ResultsMessage should render 1`] = `
<div
  className="u-flex u-flexJustifyCenter"
>
  <p
    className="ResultsMessage_text_lesmrd"
  >
    Found 
    50
     
    results
     for 
    <b>
      test
    </b>
  </p>
</div>
`;
